library/core/src/fmt/rt.rs

```diff
@@ -117,6 +117,10 @@ impl<'a> Argument<'a> {
         Self::new(x, Debug::fmt)
     }
     #[inline(always)]
+    pub fn new_debug_noop<'b, T: Debug>(x: &'b T) -> Argument<'_> {
+        Self::new(x, |_, _| Ok(()))
+    }
```

Comment:
Why are `shallow` and `none` implemented differently?
One have a separate function in the library, and another does not and is processed in `rustc_builtin_macros`.

Reply:
`none` doesn't need to print anything, so it can replace all calls with a no-op. `shallow` can't completely no-op debug print, because it's supposed to print sometimes.